pkgs/applications/terminal-emulators/mlterm/default.nix

```
@@ -42,34 +59,41 @@ stdenv.mkDerivation rec {
       --replace "-m 2755 -g utmp" " " \
       --replace "-m 4755 -o root" " "
   '';
-  NIX_LDFLAGS = "
+  NIX_LDFLAGS = lib.optionalString (!stdenv.isDarwin) "
     -L${stdenv.cc.cc.lib}/lib
     -lX11 -lgdk_pixbuf-2.0 -lcairo -lfontconfig -lfreetype -lXft
     -lvte-2.91 -lgtk-3 -lharfbuzz -lfribidi -lm17n
-  " + stdenv.lib.optionalString (openssl != null) "
+  " + lib.optionalString (openssl != null) "
     -lcrypto
-  " + stdenv.lib.optionalString (libssh2 != null) "
+  " + lib.optionalString (libssh2 != null) "
     -lssh2
   ";
 
   configureFlags = [
-    "--with-x=yes"
-    "--with-gui=xlib,fb"
     "--with-imagelib=gdk-pixbuf" #or mlimgloader depending on your bugs of choice
     "--with-type-engines=cairo,xft,xcore"
     "--with-gtk=3.0"
     "--enable-ind" #indic scripts
     "--enable-fribidi" #bidi scripts
-    "--enable-m17nlib" #character encodings
     "--with-tools=mlclient,mlconfig,mlcc,mlterm-menu,mlimgloader,registobmp,mlfc"
      #mlterm-menu and mlconfig depend on enabling gnome3.at-spi2-core
      #and configuring ~/.mlterm/key correctly.
- ] ++ stdenv.lib.optional (libssh2 == null) "--disable-ssh2";
+ ] ++ lib.optionals (!stdenv.isDarwin) [
+   "--with-x=yes"
+   "--with-gui=xlib,fb"
+    "--enable-m17nlib" #character encodings
+ ] ++ lib.optionals stdenv.isDarwin [
+    "--with-gui=quartz"
+ ] ++ lib.optional (libssh2 == null) "--disable-ssh2";
```

nitpick, let's use `optionals` here
```suggestion
 ] ++ lib.optionals (libssh2 == null) [ "--disable-ssh2" ];
```

---

I only copied this over from the original but I guess might as well change it.